llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -2507,6 +2372,24 @@ void computeKnownBits(const Value *V, const APInt &DemandedElts,
   if (isa<PointerType>(V->getType())) {
     Align Alignment = V->getPointerAlignment(Q.DL);
     Known.Zero.setLowBits(Log2(Alignment));
+    if (Q.CxtI) {
+      const Function *CxtFunc = Q.CxtI->getFunction();
+      auto InvalidUser = [&Known, &CxtFunc, &Q](const auto *Instruction) {
+        return Known.countMinTrailingZeros() >=
+                   Log2(getLoadStoreAlignment(Instruction)) ||
+               Instruction->getFunction() != CxtFunc ||
```

Comment:
I don't see why you need the first condition.
```suggestion
        return Instruction->getFunction() != CxtFunc ||
```

Reply:
See my previous comment https://github.com/llvm/llvm-project/pull/166935#discussion_r2519200403.